data/src/main/java/io/gnosis/data/models/CoinBalances.kt

```diff
@@ -0,0 +1,33 @@
+package io.gnosis.data.models
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+import pm.gnosis.common.adapters.moshi.BigDecimalNumber
+import pm.gnosis.common.adapters.moshi.DecimalNumber
+import pm.gnosis.model.Solidity
+import java.math.BigDecimal
+import java.math.BigInteger
+
+data class CoinBalances(
+    val fiatTotal: BigDecimal,
+    val items: List<Balance>
+)
+
+@JsonClass(generateAdapter = true)
+data class Balance(
+    @Json(name = "tokenInfo") val tokenInfo: TokenInfo,
+    @Json(name = "balance") @field:DecimalNumber val balance: BigInteger,
+    @Json(name = "fiatBalance") @field:BigDecimalNumber val fiatBalance: BigDecimal
+)
+
+@JsonClass(generateAdapter = true)
+data class TokenInfo(
+    @Json(name = "type") val tokenType: TokenType,
+    @Json(name = "address") val address: Solidity.Address,
+    @Json(name = "decimals") val decimals: Int = 0,
+    @Json(name = "symbol") val symbol: String = "",
+    @Json(name = "name") val name: String = "",
+    @Json(name = "logoUri") val logoUri: String?
+)
+
+enum class TokenType { ETHER, ERC20, ERC721 }
```

Comment:
If the models from the app look exactly the same as the ones from the gateway DTO , does it make sense to redefine them?

Reply:
it is cleaner to split dto's and app models. Currently, they are practically identical but this will allow more flexibility later if something has to be adjusted

Reply:
But the gateway provides objects based on the mocks done for the app's UI. This is actually redundant and confusing. If you need to add a layer of mapping it could be done at a later point. I think we also have the same issue with the transaction list where we are mapping structures that were made essentially looking at the apps UI anyway.

Reply:
I would move this discussion out of this pr. We currently do have a mapping from our dto's to models as a general approach. I personally think it makes architecture more flexible, cleaner and ready for possible future changes without introducing too much overhead. We can discuss it later in more detail and create refactoring task if needed.

Reply:
I would argue that the sole purpose of the gateway is to prevent having to do this sort of mapping. We are introducing complexity for the sake of a potential future that is not even possible as far as the purpose of the gateway is concerned.